qbraid/interface/qbraid_qasm/test/test_qasm_utils.py

```diff
@@ -48,6 +48,51 @@ def _check_output(output, expected):
     assert actual_circuit == expected_circuit
 
 
+def test__qasm_random():
+    """test random circuit"""
+
+    # Test case 1: Generate a random circuit with default parameters
+    circuit = _qasm_random()
+    assert qasm_num_qubits(circuit) >= 1
+
+    # Test case 2: Generate a random circuit with specified parameters
+    circuit = _qasm_random(num_qubits=3, depth=3, max_operands=3, seed=42)
+    assert qasm_num_qubits(circuit) == 3
+
+    # Test case 3: Generate a random circuit with measurement
+    circuit = _qasm_random(measure=True)
+    assert qasm_num_qubits(circuit) >= 1
+
+    # Test case 3: Generate a random circuit with measurement
+    circuit = _qasm_random(num_qubits=3, depth=3, max_operands=3, seed=42, measure=True)
```

Comment:
Let's split Test case 1,2,3 into three separate unit tests, or use `@pytest.mark.parametrize` to run all three tests using the same test function, but reducing redundant code.